package.json

```diff
@@ -24,6 +24,7 @@
   },
   "homepage": "https://github.com/Cimpress-MCP/AT-HttpClient#readme",
   "dependencies": {
+    "@alphatango/exceptions": "^0.0.4",
```

Comment:
The plan is to add `AT-HttpClient` and `At-OpenAPI` as `peerDependency` in `@alphatango/exceptions`

Reply:
I'm not sure I understand. Isn't it the opposite? I.e. `AT-HttpClient` and `AT-OpenAPI`  have a `peerDependency` on `@alphatango/exceptions`?

Therefore, in a project where we will be using these 2 libs, we need to provide a version of `@alphatango/exceptions`?

Reply:
I'm not sure haha! I actually thought the same before, I added the errors as peerDependency here but that won't install them. In that case we will need to have it both as a peer and normal dependency. Is this fine? I'm still trying to wrap my head around it...

On the other hand, if we add the open api and http client into errors as peer dependencies, you'll need to install specific verions of these two in the project where you install errors. It still feels weird, but it kinda reflects what are doing?

Reply:
I don't know about a good way to automatically install `peerDepedencies`. We can put them in `devDepedencies` so that they are not used in production builds, but that sounds bad as it will make development builds behave differently. I think what we would need to do is to modify CI/CD to install the missing peer deps on its own. 

----
Thinking about your proposal again. It implies we shouldn't use AT-Exceptions without the AT-HttpClient or AT-OpenApi (npm/yarn will still allow it but spill out warning of unmet peer dependency).

I have to admit that it looks weird on the first look, but it kind of makes sense as you said. The AT-Exceptions package is "useless" when it doesn't have the handling middleware specified in AT-OpenApi.

The AT-HttpClient is not required that much, e.g. there can be a service that doesn't need to use HttpClient but will still want to use AT-OpenApi and AT-Exceptions. But fine, the chance is low and it will only be a warning.

----
So the last question would be, can you think of an example of how this will work? E.g.
```
HttpClient 1.0.0:
deps: ["@alphatango/exceptions": "0.0.4"]

OpenApi 1.0.0:
deps: ["@alphatango/exceptions": "0.0.5"]

Exceptions 0.0.5:
peerDeps: ["@alphatango/httpClient": "1.0.x", "@alphatango/openApi": "1.0.x"]
```
The assumption here is that we use semantic versioning correctly and that patch versions won't break their contracts (even minor versions shouldn't). So that installing `exceptions: 0.0.5` will require us to use `1.0.x` versions of client and openapi.

Reply:
Yeah I was thinking exactly the same as in your example. The only issue is if you e.g. add something new to `@alphatango/openApi` and bump the minor version to 1.1.x , you won't be able to use the exceptions that peer depend on 1.0.x of the open api right? So we would have to update the exceptions package rather frequently even though there are no actual changes...

Maybe having the `devDependency` + `peerDependency` in the openapi & client packages would work better as we won't need to change the exceptions package a lot?

Reply:
Well just make the exceptions peer depend on `1.x` if we don't make breaking changes in minor versions and then we don't need to update the exceptions package frequently :)

Reply:
Updated the MR based on our latest discussion - added the peerDepenency on at exceptions in this project